File: java/PE041.java
public class PE041
{
    public static void main(String[] args)
    {
        int max = 0;
        for(int i = 2143; i < 9876543; i += 2) {
            if(isPrime(i) && isPandigital(i)) {
                //System.out.println(i);
                max = i;
            }
        }
        System.out.println(max);
    }
    
    private static boolean isPrime(int n) 
    {
        int k = (int)Math.sqrt(n);
        for(int i = 2; i <= k; i++) {
            if(n%i == 0)
                return false;
        }
        return true;
    }

    private static boolean isPandigital(int n) {
        int[] a = new int[10];
        for(int i = 0; i < 10; i++) 
            a[i] = 0;
        int s = 0;
        while(n != 0) {
            a[n%10]++;
            n /= 10;
            s++;
        }
        int i;
        for(i = 1; i < 10; i++) {
            if(a[i] != 1)
                break;
        }
        if(i-1 == s)
            return true;
        else
            return false;
    }
}

File: cpp/PE041.cpp
#include <iostream>
#include <cmath>
using namespace std;
bool isPrime(int n)
{
    if(n == 1)
        return false;
    else if(n < 4)
        return true;
    else if(n%2 == 0)
        return false;
    else if(n < 9)
        return true;
    else if(n%3 == 0)
        return false;
    else{
        int r = (int)sqrt((double)n);
        int f = 5;
        while(f <= r){
            if(n%f == 0)
                return false;
            else if(n%(f+2) == 0)
                return false;
            f += 6;
        }
    }
    return true;
}
int main()
{
    int max = 0, a[10], k, i, j, m;
    for(i = 2; i <= 9876543; i++){
        if(isPrime(i)){
            for(j = 0; j < 10; j++)
                a[j] = 0;
            k = 0;
            m = i;
            while(m){
                k++;
                a[m%10] ++;
                m /= 10;
            }
            for(j = 1; j <= k; j++){
                if(a[j] == 0)
                    break;
            }
            if(j == k+1 && i >= max)
                max = i;
        }
    }
    cout << max << endl;
    return 0;
}
